vowpalwabbit/parse_regressor.cc

```diff
@@ -328,18 +328,17 @@ void save_load_header(
                 model_file, (char*)&inter_len, sizeof(inter_len), "", read, msg, text);
             if (!read)
             {
-              memcpy(buff2, all.interactions[i].c_str(), inter_len);
+              memcpy(buff2, &all.interactions[i], inter_len);
```

Comment:
since interactions is a vector of vectors, you'll need to memcpy the underlying buffer of one of its elements
`memcpy(buff2, all.interactions[i].data(), inter_len);`